gpudrive/datatypes/trajectory.py

```diff
@@ -5,6 +5,39 @@
 TRAJ_LEN = 91  # Length of the logged trajectory
 
 
+def to_local_frame(
+    global_pos_xy: torch.Tensor,
+    ego_pos: torch.Tensor,
+    ego_yaw: torch.Tensor,
+    device: str,
+) -> torch.Tensor:
+    """
+    Transform trajectory from global coordinates to ego-centric frame.
+    Args:
+        global_pos_xy: Shape (time_steps, 2) containing x,y coordinates in global frame
+        ego_pos: Shape (2,) containing ego x,y position
+        ego_yaw: Shape (1,) containing ego yaw angle in radians
+    Returns:
+        transformed_trajectory: Shape (time_steps, 2) in ego-centric frame
+    """
+    # Step 1: Translate trajectory to be relative to ego position
+    translated = global_pos_xy - ego_pos
+
+    # Step 2: Rotate trajectory to align with ego orientation
+    # Create rotation matrix
+    cos_yaw = torch.cos(ego_yaw).to(device)
+    sin_yaw = torch.sin(ego_yaw).to(device)
+    rotation_matrix = torch.tensor(
+        [[cos_yaw, sin_yaw], [-sin_yaw, cos_yaw]]
+    ).to(device)
+
+    # Apply rotation matrix to the translated trajectory
+    # We need to transpose rotation_matrix for batch matrix multiplication
+    transformed_trajectory = torch.matmul(translated, rotation_matrix.T)
```

Comment:
just noting to be careful here and check a few edge cases that this transforms things correctly